ShopAppTests/App/Main/Controllers/HomeViewControllerSpec.swift

```diff
@@ -0,0 +1,111 @@
+//
+
+import Nimble
+import Quick
+
+@testable import ShopApp
+
+class HomeViewControllerSpec: QuickSpec {
+    override func spec() {
+        var viewController: HomeViewController!
+        var viewModelMock: HomeViewModelMock!
+        var tableProvider: HomeTableProvider!
+        var tableView: UITableView!
+
+        beforeEach {
+            viewController = UIStoryboard(name: "Main", bundle: nil).instantiateViewController(withIdentifier: ControllerIdentifiers.home) as! HomeViewController
+            
+            let articleRepositoryMock = ArticleRepositoryMock()
+            let articleListUseCaseMock = ArticleListUseCaseMock(repository: articleRepositoryMock)
+            let productRepositoryMock = ProductRepositoryMock()
+            let productListUseCaseMock = ProductListUseCaseMock(repository: productRepositoryMock)
+            viewModelMock = HomeViewModelMock(articleListUseCase: articleListUseCaseMock, productListUseCase: productListUseCaseMock)
+            viewController.viewModel = viewModelMock
+            
+            tableProvider = HomeTableProvider()
+            viewController.tableProvider = tableProvider
+            
+            tableView = self.findView(withAccessibilityLabel: "tableView", in: viewController.view) as! UITableView
+            
+            _ = viewController.view
+        }
+        
+        describe("when view loaded") {
+            beforeEach {
+                viewController.viewWillAppear(false)
+            }
+            
+            it("should have correct superclass") {
+                expect(viewController).to(beAKindOf(BaseTableViewController<HomeViewModel>.self))
+            }
+            
+            it("should have correct table provider class") {
+                expect(viewController.tableProvider).to(beAnInstanceOf(HomeTableProvider.self))
+            }
+            
+            it("should have correct delegate of table provider") {
+                expect(tableProvider.delegate) === viewController
+            }
+            
+            it("should have correct data source and delegate of table view") {
+                expect(tableView.dataSource) === tableProvider
+                expect(tableView.delegate) === tableProvider
+            }
+            
+            it("should have correct table view insets") {
+                expect(tableView.contentInset) == TableView.defaultContentInsets
+            }
+            
+            it("should have correct title") {
+                expect(viewController.navigationItem.title) == "ControllerTitle.Home".localizable
+            }
+            
+            it("should have cart button") {
+                expect(viewController.navigationItem.rightBarButtonItem).toNot(beNil())
+            }
+            
+            it("should start loading data") {
+                expect(viewModelMock.isDataLoadingStarted) == true
+            }
+        }
+        
+        describe("when data loaded") {
```

Comment:
I think we need to test case when there are no articles and table view has two sections instead.